frontend/ui/lib/widgets/create_trip_widget.dart

```diff
@@ -26,13 +26,12 @@ class _CreateTripWidgetState extends State<CreateTripWidget> {
   String place;
   int radius = 0;
   String newInformation = 'Grabbed info placed here.';
-  void submitTrip() {
-    setState(() {
-      tripService.createTrip(Trip(
+  void submitTrip() async {
+    await tripService.createTrip(Trip(
```

Comment:
To avoid leaking an exception and breaking the client, you should run this in a try-catch block. In the case of an exception, you can either log something to the console or [display a snackbar](https://flutter.dev/docs/cookbook/design/snackbars) (preferable).

Reply:
I recommend adding a "dismiss" action as well to make it easier for users to dismiss the warning. Here's what it might look like:

```dart
    try {
      await tripService.createTrip(Trip(
          name: place,
          id: radius.toString(),
          location: Location(latitude: 10.0, longitude: 10.0)));
      setState(() {
        newInformation = "$place ${radius}km";
      });
    } catch (e, s) {
      print("error creating trip: $e");
      print(s);
      Scaffold.of(context).showSnackBar(SnackBar(
        content: Text("Error creating trip"),
        action: SnackBarAction(
          label: "Dismiss",
          onPressed: () {},
        ),
      ));
    }
```

Note that we'll eventually want to submit the real trip data as entered on this page.